lib/org/chromium/webidl/Canonicalizer.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'Canonicalizer',
+
+  documentation: `This component expects to receive AST nodes from various
+    sources. Canonical versions of interface definitions will be created
+    from AST nodes of the same source.
+    Please note that this step cannot be distributed. All computations
+    must be done through one canonicalizer instances, or you will
+    get fragmented results.`,
+
+  requires: [
+    'org.chromium.webidl.ast.Enum',
+    'org.chromium.webidl.ast.Typedef',
+    'org.chromium.webidl.PipelineMessage',
+  ],
+
+  properties: [
+    {
```

Comment:
Please add a `TODO`. This component should not be responsible for observing the pipeline; eventually `onDone` and `waitTime` should be removed or replace with a subscription to a component that *is* responsible for observing the pipeline.